README.md

```diff
@@ -1458,6 +1479,233 @@ map.addEntryListener(mapEventListener).then(function () {
+- `name NOT LIKE 'Jo_'` (true for 'Josh'; false for 'Joe')
+- `name LIKE 'J_s%'` (true for 'Josh', 'Joseph'; false 'John', 'Joe')
+
+**ILIKE:** `<attribute> [NOT] ILIKE 'expression'`
+
+Similar to LIKE predicate but in a case-insensitive manner.
+
+- `name ILIKE 'Jo%'` (true for 'Joe', 'joe', 'jOe','Josh','joSH', etc.)
+- `name ILIKE 'Jo_'` (true for 'Joe' or 'jOE'; false for 'Josh')
+
+**REGEX:** `<attribute> [NOT] REGEX 'expression'`
+
+- `name REGEX 'abc-.*'` (true for 'abc-123'; false for 'abx-123')
+
+##### Querying Entry Keys with Predicates
+
+You can use `__key` attribute to perform a predicated search for entry keys. Please see the following example:
+
+```javascript
+var personMap = client.getMap('persons');
+personMap.put('Ahmet', new Person('Ahmet', 27)).then(function () {
+    return personMap.put('Ali', new Person('Ali', 30));
+}).then(function () {
+    return personMap.put('Furkan', new Person('Furkan', 23));
+}).then(function () {
+    var predicate = new Predicates.sql('__key like A%');
+    return personMap.valuesWithPredicate(predicate);
+}).then(function (startingWithA) {
+    // some operations
+});
+```
+
+In this example, the code creates a list with the values whose keys start with the letter "A”.
+
+#### 1.4. Filtering with Paging Predicates
+
+The Node.js client provides paging for defined predicates. With its `PagingPredicate` object, you can get a list of keys, values, or entries page by page by filtering them with predicates and giving the size of the pages. Also, you can sort the entries by specifying comparators.
+
+```javascript
+var map = hazelcastClient.getMap('students');
+var greaterEqual = Predicates.greaterEqual('age', 18);
+var pagingPredicate = Predicates.paging(greaterEqual, 5);
+
+// Set page to retrieve third page
+pagingPredicate.setPage(3);
+
+// Retrieve third page
+map.valuesWithPredicate(pagingPredicate).then(function (values) {
+    // some operations
+    ...
+    
+    // Set up next page
+    pagingPredicate.nextPage();
+    
+    // Retrieve next page
+    return map.valuesWithPredicate(pagingPredicate);
+}).then(function (values) {
+    // some operations
+});
+```
+
+If you want to sort the result before paging, you need to specify a comparator object that implements `Comparator` interface. Also, this comparator object should be one of `IdentifiedDataSerializable` or `Portable`. After implementing the Node.js version, you need to implement the Java equivalent of the comparator and its factory. The Java equivalent of the comparator should implement `java.util.Comparator` interface's `compare` method which will specify how to sort the result. Note that `comprare` function of `Comparator` on the Java side is the equivalent of the `sort` function of `Comparator` on the Node.js side. When you implement `Comparator` and its factory, you can add them to the `CLASSPATH` of the server side.  See [Adding User Library to CLASSPATH section](#adding-user-library-to-classpath).
```

Comment:
"..should implement `java.util.Comparator` interface's `compare` method which will specify how to sort the result." -> ".. should implement `java.util.Comparator`"